numba/_devicearray.cpp

```diff
@@ -85,10 +85,31 @@ PyTypeObject DeviceArrayType = {
     0,                                           /* tp_del */
     0,                                           /* tp_version_tag */
     0,                                           /* tp_finalize */
-#if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION == 8
+/* The docs suggest Python 3.8 has no tp_vectorcall
+ * https://github.com/python/cpython/blob/d917cfe4051d45b2b755c726c096ecfcc4869ceb/Doc/c-api/typeobj.rst?plain=1#L146
+ * but the header has it:
```

Comment:
😮

Given how crazy it is already, would it actually worth moving to the [PyType_FromSpec](https://pythoncapi.readthedocs.io/type_object.html#pytype-fromspec) API so we never have to change this again?

Reply:
I think that this could potentially work and would certainly have the noted advantage. I've done some preliminary work in this area and the code changes are unfortunately much larger than the scope of this PR, so have deferred this to #8669.